report-viewer/src/model/factories/OverviewFactory.ts

```diff
@@ -35,33 +35,41 @@ export class OverviewFactory {
     const map = new Map<string, string>(Object.entries(jsonSubmissions))
     const dateOfExecution = json.date_of_execution as string
     const duration = json.execution_time as number as number
-    const metrics = [] as Array<Metric>
     const clusters = [] as Array<Cluster>
     const totalComparisons = json.total_comparisons as number
-    ;(json.metrics as Array<unknown>).forEach((jsonMetric) => {
-      const metric = jsonMetric as Record<string, unknown>
-      const comparisons = [] as Array<ComparisonListElement>
-
-      ;(metric.topComparisons as Array<Record<string, unknown>>).forEach(
-        (jsonComparison, index) => {
-          const comparison: ComparisonListElement = {
-            id: (index + 1) as number,
-            firstSubmissionId: jsonComparison.first_submission as string,
-            secondSubmissionId: jsonComparison.second_submission as string,
-            similarity: jsonComparison.similarity as number
-          }
-          comparisons.push(comparison)
-        }
+
+    const distributions = [] as Array<Distribution>
+    const averageSimilarities: Map<string, number> = new Map<string, number>()
+    const comparisons = [] as Array<ComparisonListElement>
+
+    const metrics = json.metrics as Array<unknown> as Array<Record<string, unknown>>
+    // Average
```

Comment:
Could be split up in functions

Reply:
I would put his into a new PR since we are planning on changing something in the json to make it easier to implement new metrics. Then this code will have to change anyways

Reply:
Yes, please regard all of my comments as things that could also be done in an additional PR to speed up the process :)